contracts/protocol/pool/PoolApeStaking.sol

```diff
@@ -0,0 +1,376 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.10;
+
+import "../libraries/paraspace-upgradeability/ParaReentrancyGuard.sol";
+import "../libraries/paraspace-upgradeability/ParaVersionedInitializable.sol";
+import {PoolStorage} from "./PoolStorage.sol";
+import "../../interfaces/IPoolApeStaking.sol";
+import "../../interfaces/IPToken.sol";
+import "../../dependencies/yoga-labs/ApeCoinStaking.sol";
+import "../../interfaces/IXTokenType.sol";
+import "../../interfaces/INTokenApeStaking.sol";
+import {ValidationLogic} from "../libraries/logic/ValidationLogic.sol";
+import {IPoolAddressesProvider} from "../../interfaces/IPoolAddressesProvider.sol";
+import {Errors} from "../libraries/helpers/Errors.sol";
+import {WadRayMath} from "../libraries/math/WadRayMath.sol";
+import {ReserveLogic} from "../libraries/logic/ReserveLogic.sol";
+import {UserConfiguration} from "../libraries/configuration/UserConfiguration.sol";
+import {ApeStakingLogic} from "../tokenization/libraries/ApeStakingLogic.sol";
+
+contract PoolApeStaking is
+    ParaVersionedInitializable,
+    ParaReentrancyGuard,
+    PoolStorage,
+    IPoolApeStaking
+{
+    using ReserveLogic for DataTypes.ReserveData;
+    using UserConfiguration for DataTypes.UserConfigurationMap;
+    using SafeERC20 for IERC20;
+
+    IPoolAddressesProvider internal immutable ADDRESSES_PROVIDER;
+    uint256 internal constant POOL_REVISION = 1;
+
+    /**
+     * @dev Emitted on setUserUseERC20AsCollateral()
+     * @param reserve The address of the underlying asset of the reserve
+     * @param user The address of the user enabling the usage as collateral
+     **/
+    event ReserveUsedAsCollateralEnabled(
+        address indexed reserve,
+        address indexed user
+    );
+
+    /**
+     * @dev Emitted on setUserUseERC20AsCollateral()
+     * @param reserve The address of the underlying asset of the reserve
+     * @param user The address of the user enabling the usage as collateral
+     **/
+    event ReserveUsedAsCollateralDisabled(
+        address indexed reserve,
+        address indexed user
+    );
+
+    /**
+     * @dev Constructor.
+     * @param provider The address of the PoolAddressesProvider contract
+     */
+    constructor(IPoolAddressesProvider provider) {
+        ADDRESSES_PROVIDER = provider;
+    }
+
+    function getRevision() internal pure virtual override returns (uint256) {
+        return POOL_REVISION;
+    }
+
+    function depositApeCoin(
+        address nftAsset,
+        ApeCoinStaking.SingleNft[] calldata _nfts
+    ) external nonReentrant {
+        DataTypes.PoolStorage storage ps = poolStorage();
+
+        DataTypes.ReserveData storage nftReserve = ps._reserves[nftAsset];
+        INToken nToken = INToken(nftReserve.xTokenAddress);
+        require(
+            nToken.getXTokenType() == XTokenType.NTokenBAYC ||
+                nToken.getXTokenType() == XTokenType.NTokenMAYC,
+            Errors.INVALID_ASSET_TYPE
```

Comment:
assetType means ERC20 | ERC721, this may be a bit confusing. Or we can remove UNIV3_NOT_ALLOWED and SAPE_NOT_ALLOWED